feat(rh-timestamp): add rh-timestamp

eyevana opened the pull request:
Closes #548

## What I did

1. Created a new element, `rh-timestamp`, based on `pf-timestamp`.
3. Fixed a type in the `feedback` partial that was returning the wrong markup for element pages.
4. Added docs


## Testing Instructions

1. Check out the demos and verify the examples are working as expected.
2. Verify docs match what Corey provided in #548.

bennypowers commented on elements/rh-timestamp/demo/rh-timestamp.html:
```diff
@@ -1,4 +1,90 @@
 <link rel="stylesheet" href="demo.css">
 <script type="module" src="rh-timestamp.js"></script>
 
-<rh-timestamp></rh-timestamp>
```
we'd prefer to split these out into separate demos. you can keep a `kitchen-sink.html` is that's easier for dev. see alert demos

eyevana replied:
I see what you mean. Will do! :)

bennypowers commented on docs/cheatsheet.njk:
can we revert these unrelated changes and put them in a new PR? maybe you just need to rebase?

bennypowers reviewed (changes requested):
needs a minor changeset. please copy an example from the changelog to maintain the format

eyevana commented:
@coreyvickery timestamp docs are ready for review. Check it out :) https://deploy-preview-1149--red-hat-design-system.netlify.app/elements/

eyevana commented on elements/rh-timestamp/README.md:
@bennypowers do we still need this file? I'm asking because I noticed the latest element added to RHDS didn't have one.

zeroedin replied:
I believe we should still have this file even if with the components most basic usage and maybe a link to both docs and demo of the component.   As an example secondary nav's README is maybe overkill?

A more general reply and not specific to this PR alone is that we should ensure we have the same info in docs/demos/ and the README in sync such as component description / usage demo code.  Wondering if there is a better way to do this, maybe build these files automation pulling from the other sources we already maintain.

bennypowers commented on .changeset/strong-pens-boil.md:
````diff
@@ -0,0 +1,11 @@
+  Provides consistent formats for displaying date and time values.
+
+  ```html
+  <rh-timestamp></rh-timestamp>
````
This needs an attribute?

bennypowers commented on docs/scss/components/_example.scss:
```diff
@@ -66,6 +66,10 @@ rh-alert + .example {
     }
   }
 
+  &--component {
```
Can you do this with using sass concat or bem? We're trying to move away from those

bennypowers commented on elements/rh-timestamp/README.md:
````diff
@@ -0,0 +1,13 @@
+Use a timestamp to display date and time values.
+
+```html
+<rh-timestamp></rh-timestamp>
````
Attr?

bennypowers replied:
Attr and Gainesville content?

eyevana commented:
@bennypowers thank you so much for the review! <3 I removed the BEM modifier (sorry, old habit), added some examples to the main demo file so users can preview in the playground, and added a specific date attribute to those examples you pointed out instead of defaulting to the current date. Let me know if you catch anything else.

bennypowers commented on .changeset/strong-pens-boil.md:
````diff
@@ -0,0 +1,11 @@
+  Provides consistent formats for displaying date and time values.
+
+  ```html
+  <rh-timestamp date="Tue Aug 09 2006 14:57:00 GMT-0400"></rh-timestamp>
````
```suggestion
  <rh-timestamp date="Tue Aug 09 2006 14:57:00 GMT-0400">Tue Aug 09 2006 14:57:00 GMT-0400</rh-timestamp>
```

bennypowers commented on elements/rh-timestamp/rh-timestamp.ts:
```diff
@@ -0,0 +1,115 @@
+    }
+  }
+
+  willUpdate(changedProperties: PropertyValues<this>) {
```
```suggestion
  override willUpdate(changedProperties: PropertyValues<this>) {
```

bennypowers commented on elements/rh-timestamp/rh-timestamp.ts:
```diff
@@ -0,0 +1,115 @@
+
+  #timestamp = new TimestampController(this);
+
+  connectedCallback() {
```
```suggestion
  override connectedCallback() {
```

bennypowers commented on elements/rh-timestamp/rh-timestamp.ts:
```diff
@@ -0,0 +1,115 @@
+    }
+  }
+
+  render() {
```
```suggestion
  override render() {
```

bennypowers reviewed (approved):
last couple tweaks but lgtm
